src/App.jsx

```diff
@@ -7,45 +7,45 @@ import "./pages/Detail.css";
 import SearchPage from "./pages/SearchPage";
 import SearchDetail from "./pages/SearchDetail";
 import Login from "./pages/Login";
+import { Device } from "@deco3801-mortein/mortein-sdk/services.gen";
 
 function App() {
     const [userData, setUserData] = useState(null);
 
-    // Function to toggle vibration
-    function toggleVibration(event) {
-        let newDevices = [];
-        for (let i = 0; i < userData.devices.length; i++) {
-            let device = userData.devices[i];
-            if (device.id === event.target.value) {
-                device = {
-                    ...device,
-                    data: {
-                        ...device.data,
-                        vibration: !device.data.vibration,
-                    },
-                };
-            }
-            newDevices.push(device);
-        }
-        setUserData((prevData) => ({
-            ...prevData,
-            devices: newDevices,
-        }));
-    }
-
-    // Fetch API data
     useEffect(() => {
-        fetch("http://localhost:3000/users/1")
-            .then((response) => response.json())
+        const devices = Device.getDevice();
+        devices.promise
```

Comment:
`devices.then` abstracts the need to access `devices.promise.then`

```suggestion
        devices
```

This suggestion applies in a few other places too.